openedx/features/colaraz_features/views.py

```diff
@@ -37,3 +49,85 @@ def get_redirect_url(self, *args, **kwargs):
                     return req.url
 
         return reverse('logout')
+
+
+class BaseViewMixin(object):
+    """
+    Mixin for setting template engine and enforcing authentication before page view.
+    """
+    template_engine = Engines.MAKO
+    permission_classes = (IsAdminOrOrganizationalInstructor, )
+
+    @classmethod
+    def as_view(cls, **initkwargs):
+        """
+        Override `as_view` to perform login check.
+        """
+        return login_required(
```

Comment:
Can we achieve this behavior by using the `IsAuthenticated` permission class?

Reply:
Yes, I believe it can be used. I will update it.

Reply:
@danialmalik I tried using `IsAuthenticated` permission class here. but the way permissions are implemented. URL state is not stored and if user is not authenticated and is redirected to the login page then after login he is not redirected back to the original page, instead, he ends up in dashboard page.

So, I have kept the code utilizing `login_required` as it offers better UX.